src/plugins/intel_npu/src/compiler/include/graph_transformations.hpp

```diff
@@ -14,9 +17,46 @@
  */
 namespace intel_npu::driverCompilerAdapter {
 
-/**
- * @brief Serialize OpenVINO model to IR
- */
-IR serializeToIR(const std::shared_ptr<const ov::Model>& model, uint32_t supportedVersionByCompiler = 7);
+class IR {
```

Comment:
This is a helper used in a single function, it does not have a clear semantics nor a clear name.
Do we even need that class? Consider doing that in the function.
or a function template if we really need to keep two implementations.

If we must keep both better make abstraction over a single stream and then compose it, abstraction for two streams leads to duplicate code (same code/variables for each stream).

Reply:
We have struct IR in old version, the reason to upgrade to class is to wrap the logical to choose stream. As I mentioned before, we still not sure which is best solution for small model | large model | model larger than memory, so only use custom stream as corner case for large model on windows now. 
Use template will move the logic which choose stream kind to other file, I think package serialization related logics to graph_transformers.hpp make it easy to maintain now.